src/main/java/org/opensearch/security/configuration/DlsFlsValveImpl.java

```diff
@@ -558,12 +562,11 @@ private void setFlsHeaders(EvaluatedDlsFlsConfig dlsFls, ActionRequest request)
                 }
             } else {
                 if (threadContext.getHeader(ConfigConstants.OPENDISTRO_SECURITY_FLS_FIELDS_HEADER) != null) {
-                    if (!flsFields.equals(
-                        Base64Helper.deserializeObject(
-                            threadContext.getHeader(ConfigConstants.OPENDISTRO_SECURITY_FLS_FIELDS_HEADER),
-                            threadContext.getTransient(ConfigConstants.USE_JDK_SERIALIZATION)
-                        )
-                    )) {
+                    Map<String, Set<String>> deserializedMap = (Map<String, Set<String>>) Base64Helper.deserializeObject(
+                        threadContext.getHeader(ConfigConstants.OPENDISTRO_SECURITY_FLS_FIELDS_HEADER),
+                        threadContext.getTransient(ConfigConstants.USE_JDK_SERIALIZATION)
+                    );
+                    if (!isSubMap(flsFields, deserializedMap)) {
```

Comment:
This is for clusters with mixed OpenSearch versions, right?

I think there's a catch: This will work in the case that a request first hits a node with an old OpenSearch version which then gets forwarded to a node with a newer OpenSearch version. 

However, in the reverse case, there will be trouble: If a request first hits an OpenSearch version with this change, this node will calculate the filtered map and pass it on. If the request is then passed on to an older OpenSearch version, it will calculate a different map and throw an exception.

A different perspective: TBH, it is totally unclear to me what's actually the purpose of this equality check. So far, I could not think of any real purpose. I would have the tendency to say that this code could be removed alltogether. (Still, this would not fix the mixed cluster issue directly).

Reply:
Added a check for min node version in. If the cluster has nodes <= 2.17 it will place in the full map as expected. If the min version is 2.18 or beyond it will place the filtered map in as headers.